### DISCUSSION TITLE
How to get gui:get_appearance() to work?

### DISCUSSION THREAD

**Discussion by robrecord** [wezterm/wezterm#3138] [Q&A]
I've followed instructions to have my theme change depending on whether macOS is using dark mode or not, and this works well. I've now tried to use the same code within a required file to set up some tab bar colours differently, but the get_appearance function is returning Light when it's actually Dark, only in this case. The query is happening very early on in the file and not inside any hook. I'm wondering if there's a point before which wezterm has not been able to get the os appearance, and if my code should compensate for this by querying it later eg when the tab bar triggers are happening? Or is it a bug?

**Comment by wez** [comment]
If you run `WEZTERM_LOG=window::os::macos::connection=debug,info wezterm` it should log some lines about `NSAppearanceName` to the logs; can you share those?

**Comment by robrecord** [comment]
The issue seems to be resolved somehow. Thanks anyway wez. If it happens again I'll do those logs for you.
